CMakeLists.txt

```diff
@@ -19,6 +19,12 @@ if(NOT CMAKE_CXX_STANDARD MATCHES "^(17|20)$")
   message(FATAL_ERROR "CMAKE_CXX_STANDARD must be set to 17 or 20")
 endif()
 
+# Prevent unnecessary re-linking on Linux
+# See https://discourse.slicer.org/t/build-when-git-commit-is-changed/29607/12
```

Comment:
Provide more information about what kind of relinking is avoided. Provide a specific example (e.g., changing X file causes relinking of Y library), because that allows testing in the future if the flag is still needed.